v3-nginx/docker-entrypoint.d/91-update-resolver.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+# vim:sw=2:ts=2:sts=2:et
+
+set -eu
+
+LC_ALL=C
+ME=$( basename "$0" )
+PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+
+DNS_SERVER="${DNS_SERVER:-$(grep -i '^nameserver' /etc/resolv.conf | head -n1 | cut -d ' ' -f2)}"
+
+sed -i.bak -r 's/DNS_SERVER/'"${DNS_SERVER}"'/' /etc/nginx/nginx.conf
```

Comment:
According to the man page, `-E` should be used for portability. Also, I think you can use a single string to allow for the variable expansion, since there are no shell characters in there.

```suggestion
sed -i.bak -E "s/DNS_SERVER/${DNS_SERVER}/" /etc/nginx/nginx.conf
```